ddl/column.go

```diff
@@ -1733,9 +1733,19 @@ func isColumnWithIndex(colName string, indices []*model.IndexInfo) bool {
 	return false
 }
 
-func isColumnCanDropWithIndex(colName string, indices []*model.IndexInfo) bool {
+func isColumnCanDropWithIndex(isMultiSchemaChange bool, colName string, indices []*model.IndexInfo) bool {
+	var condition func(info *model.IndexInfo) bool
+	if isMultiSchemaChange {
+		condition = func(idx *model.IndexInfo) bool {
+			return len(idx.Columns) > 1
+		}
+	} else {
+		condition = func(idx *model.IndexInfo) bool {
+			return len(idx.Columns) >= 1
+		}
+	}
 	for _, indexInfo := range indices {
-		if indexInfo.Primary || len(indexInfo.Columns) > 1 {
+		if indexInfo.Primary || condition(indexInfo) {
```

Comment:
How about 
```go
if indexInfo.Primary || len(indexInfo.Columns) > 1) || 
(!enabledMultiSchemaChange && len(indexInfo.Columns) == 1) {
```
and remove those anonymous functions?

Reply:
good